vllm/engine/arg_utils.py

```diff
@@ -667,6 +673,49 @@ def create_engine_config(self, ) -> EngineConfig:
             ray_workers_use_nsight=self.ray_workers_use_nsight,
             distributed_executor_backend=self.distributed_executor_backend)
 
+        # if use_vmm, we need to calculate the block size
+        if self.use_vmm:
+            if self.kv_cache_dtype == "auto":
+                dtype = model_config.dtype
+            else:
+                dtype = STR_DTYPE_TO_TORCH_DTYPE[self.kv_cache_dtype]
+            dtype_size = get_dtype_size(dtype)
+
+            head_size = model_config.get_head_size()
+            num_heads = model_config.get_num_kv_heads(parallel_config)
+            num_layers = model_config.get_num_layers(parallel_config)
+
+            single_token_bytes_size = head_size * num_heads * dtype_size
+            # We can divide a block equally among all layers, which reduces
+            # the number of vmm memory operations.
+            single_token_bytes_size *= num_layers
+            # vmm only support flash-attn now, which need block_size % 16 == 0
+            min_block_size = single_token_bytes_size * 16
```

Comment:
Is that inaccurate since `single_token_bytes_size` may be divided by 2/4/8/16 ? 
for example, in llama 13B, `5120 * 40 * sizeof(fp16) = 400kB`, so we don't need to multiply it by 16, right?